### PR TITLE
Exams by Tkachenko Zenoviia IM-31

### PR DESCRIPTION
**Pull request by Zenoviia**

### CODE REVIEW

File: Tasks/day.js [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
-  for (i = 0; i < D.length; i++) {
-    if (s.startsWith(D[i].toLowerCase())) {
-      return i + 1;
-    }
+// Step 5
+// Add an if-else block to improve readability
+
+'use strict';
+
+const days = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
```
Comment (tshemsedinov) [inline Tasks/day.js:8]: Naming for global constants: UPPER_SNAKE

giphy

File: Tasks/day.js [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
+
+let parseDay = (string) => {
```
Comment (tshemsedinov) [inline Tasks/day.js:10]: const

170203-salt-bae-mn-1530

File: Tasks/filter.js [hunk 3]
```diff
@@ -1,15 +1,13 @@
...
-      remove.unshift(x);
-    }
-  }
-  for (x of remove) T.splice(x, 1);
-  return T;
+// Step 3
+// Replace the loop with the built-in filter() method, and also use filter() to create an array whose elements correspond to filterType
+'use strict';
+
+'use strict';
```
Comment (tshemsedinov) [inline Tasks/filter.js:7]: - Such much?
- For whom how

ezgif-1-cc7028f68d

File: Tasks/day.js [hunk 4]
```diff
@@ -1,15 +1,19 @@
...
+  const index = days.findIndex(day => string.startsWith(day.toLowerCase()));
+  if (index !== -1) {
+    return index + 1;
+  } else {
+    return -1;
```
Comment (tshemsedinov) [inline Tasks/day.js:15]: We do not need else here, by the the way it is a good place for ternary

I9pw

File: Tasks/filter.js [hunk 5]
```diff
@@ -1,15 +1,13 @@
...
+
+const FilterByType = (array, filterType) => {
```
Comment (tshemsedinov) [inline Tasks/filter.js:9]: camelCase

giphy (1)

File: Tasks/intersection.js [hunk 6]
```diff
@@ -1,25 +1,23 @@
...
-      }
-    }
-    {
+// Step 6
+// Use forEach to iterate  the keys
+// Use the arrow function
+
+'use strict';
+
+'use strict';
```
Comment (tshemsedinov) [inline Tasks/intersection.js:9]: 1_gV-RSNes1lysqxug-_mHIQ

File: Tasks/intersection.js [hunk 7]
```diff
@@ -1,25 +1,23 @@
...
+
+const intersection = function intersection(object1, object2) {
+  const firstKeys = Object.keys(object1);
+
+  firstKeys.forEach((key) => {
```
Comment (tshemsedinov) [inline Tasks/intersection.js:14]: for..of

File: Tasks/intersection.js [hunk 8]
```diff
@@ -1,25 +1,23 @@
...
+    if (object1[key] !== object2[key]) {
+      delete object1[key];
```
Comment (tshemsedinov) [inline Tasks/intersection.js:16]: Try to implement without delete

0_4p1i8-hYPr-u4Sd0

File: Tasks/intersection.js [hunk 9]
```diff
@@ -1,25 +1,23 @@
...
     }
-  }
-  {
-  }
-  return object_1;
+  });
+
+  return object1;
```
Comment (tshemsedinov) [inline Tasks/intersection.js:20]: Never return incoming arguments

vn5zQ5

File: Tasks/intersection.js [hunk 10]
```diff
@@ -1,25 +1,23 @@
...
-    {
+// Step 6
+// Use forEach to iterate  the keys
+// Use the arrow function
+
+'use strict';
+
+'use strict';
+
+const intersection = function intersection(object1, object2) {
```
Comment (tshemsedinov) [inline Tasks/intersection.js:11]: Arrow function is preferred here

1_t50xYVs1uJb_KKErX2AMlQ

File: Tasks/intersection.js (hunk 6 above)
Comment (tshemsedinov) [reply Tasks/intersection.js:9]: Also do not change incoming args oi functions

File: Tasks/longest.js [hunk 11]
```diff
@@ -1,28 +1,23 @@
...
-        } else {
-          i = -1;
-        }
-      }
+// Step 6
+// Remove Object.assign
+
+'use strict';
+
+const longest = function (line = []) {
```
Comment (tshemsedinov) [inline Tasks/longest.js:8]: arrow function

c++-C-perl-python-2681345

File: Tasks/month.js [hunk 12]
```diff
@@ -1,6 +1,13 @@
...
 
-Months = [
+// Step 3
+// Use findIndex to find the month index
+// Save the string in lower case by creating the inputLowerCase variable
+// use the ternary operator instead of if-else
+
+'use strict';
+
+const months = [
```
Comment (tshemsedinov) [inline Tasks/month.js:10]: Use UPPER_CAMEL for global constants

free_PNG90784

File: Tasks/only.js [hunk 13]
```diff
@@ -1,16 +1,19 @@
...
+  for (const key in obj) {
+    if (!keysToKeep.includes(key)) {
+      delete obj[key];
     }
-  }, 99);
-  [].sort();
-  return W;
+  }
+
+  return obj;
```
Comment (tshemsedinov) [inline Tasks/only.js:16]: Do not mutate and return incoming args

c++-C-perl-python-2681345

File: Tasks/password.js [hunk 14]
```diff
@@ -1,13 +1,19 @@
...
 
-let GeneratePassword = (alphabet, length) => {
-  const MAX = alphabet.length;
-  key = '';
+// Step 4
+// Add an empty line for readability
+
+'use strict';
+
+let generatePassword = (alphabet, length) => {
```
Comment (tshemsedinov) [inline Tasks/password.js:8]: Use const Luke!

File: Tasks/password.js [hunk 15]
```diff
@@ -1,13 +1,19 @@
...
+  const alphabetLength = alphabet.length;
+  let key = '';
+
   for (let i = 0; i < length; i++) {
-    Index = Math.floor(Math.random() * MAX);
-    key = key + alphabet[Index];
+    let index = Math.floor(Math.random() * alphabetLength);
```
Comment (tshemsedinov) [inline Tasks/password.js:13]: Extract random as a separate func

21007cee855884a1cb9b9059151c163c

File: Tasks/sample.js [hunk 16]
```diff
@@ -1,8 +1,12 @@
...
-  arr = arr[Math.floor(Math.random() * arr.length)];
-  return arr;
+// Step 4
+// Remove the second return array as it becomes unreadable after the previous step
+
+'use strict';
+
+const sample = (arr) => {
+  return arr[Math.floor(Math.random() * arr.length)];
 };
```
Comment (tshemsedinov) [inline Tasks/sample.js:10]: Hey, it's liner

IRtk

File: Tasks/only.js [hunk 17]
```diff
@@ -1,16 +1,19 @@
...
+// Step 5
+// Save the for...in loop to repeat the properties of the object
+// Modify the original object using delete
+
+'use strict';
+
+const only = (obj, ...keysToKeep) => {
+  for (const key in obj) {
+    if (!keysToKeep.includes(key)) {
+      delete obj[key];
```
Comment (tshemsedinov) [inline Tasks/only.js:12]: Try not to use delete

no-screaming

File: Tasks/month.js [hunk 18]
```diff
@@ -12,15 +19,13 @@ Months = [
   'sep',
   'oct',
   'nov',
-  'dec',
+  'dec'
```
Comment (tshemsedinov) [inline Tasks/month.js:22]: We need comma here

magic-meme

File: Tasks/longest.js [hunk 19]
```diff
@@ -1,28 +1,23 @@
...
+  for (let i of line) {
+    if (i.length > maxLength) {
+      maxLength = i.length;
+      longestStr = i;
     }
   }
-  Object.assign(line, { length: s.length });
-  return s;
+
+  line.length = longestStr.length;
```
Comment (tshemsedinov) [inline Tasks/longest.js:19]: Never used

758A

**Comment by tshemsedinov** [comment]
1_t50xYVs1uJb_KKErX2AMlQ

File: Tasks/month.js (hunk 12 above)
Comment (Kabadzh0b) [reply Tasks/month.js:10]: UPPER_SNAKE?

> Use UPPER_CAMEL for global constants
> 
> free_PNG90784

**Pull request closed without merging**